src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,225 @@
+        this.setContent(this._badgeContent, this.badgeMaxValue);
+        this._renderer.appendChild(this._element.nativeElement, this._badgeElement);
+        this._renderer.removeStyle(this._badgeElement, 'display');
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        // if the badge is visible set changed values
+        if (this._badgeElement) {
+            // set badge content and get display friendly version of text based on max length and type of val
+            if (changes.badgeContent || changes.badgeMaxValue) {
+                let finalText: string = (changes.badgeContent && changes.badgeContent.currentValue) || this.badgeContent || null;
+                const maxValue: number = (changes.badgeMaxValue && changes.badgeMaxValue.currentValue) || this.badgeMaxValue || null;
+                this.setContent(finalText, maxValue);
+            }
+
+            // set the badge color
+            if (changes.badgeColor && changes.badgeColor.currentValue !== changes.badgeColor.previousValue) {
+                this.setBadgeColor();
+            }
+
+            // set the badge border color
+            if (changes.badgeBorderColor && changes.badgeBorderColor.currentValue !== changes.badgeBorderColor.previousValue) {
+                this.setBadgeBorderColor();
+            }
+
+            // set badge size
+            if (changes.badgeSize && changes.badgeSize.currentValue !== changes.badgeSize.previousValue) {
+                this.setBadgeSize(changes.badgeSize.previousValue);
+            }
+        }
+    }
+
+    ngOnDestroy(): void {
+        if (this._renderer.destroyNode) {
+            this._renderer.destroyNode(this._badgeElement);
+        }
+    }
+
+    private setContent(finalText: string, maxValue: number): void {
+        if (finalText && maxValue && maxValue > 0) {
+            if (this._isNumber && parseInt(finalText) > maxValue) {
+                finalText = `${maxValue}+`;
+            } else if (finalText.length > maxValue) {
+                finalText = `${finalText.substr(0, maxValue)}…`;
+            }
+        }
+
+        this._badgeDisplayContent = finalText;
+        this._badgeElement.textContent = this._badgeDisplayContent;
+    }
+
+    private setBadgeColor(): void {
+        if (this._badgeColor) {
+            this._renderer.setStyle(this._badgeElement, 'background', this._badgeColor.toHex());
+        } else {
+            this._renderer.removeStyle(this._badgeElement, 'background');
+        }
+
+        this._renderer.setStyle(this._badgeElement, 'color', this.determineContentTextColor().toHex());
+    }
+
+    private setBadgeBorderColor(): void {
+        if (this._badgeBorderColor) {
+            this._renderer.setStyle(this._badgeElement, 'border-color', this._badgeBorderColor.toHex());
+        } else {
+            this._renderer.removeStyle(this._badgeElement, 'border-color');
+        }
+    }
+
+    private setBadgeSize(previousValue: string): void {
```

Comment:
Can this parameter be made optional instead of having to pass `null`. Perhaps `previousSize` might also be clearer.